distributed/worker.py

```diff
@@ -1138,6 +1138,9 @@ async def close(
             for pc in self.periodic_callbacks.values():
                 pc.stop()
             with suppress(EnvironmentError, TimeoutError):
+                if self._client:
+                    await self._client.close()
```

Comment:
Beware that the Worker may share this client with other workers or clients in the same process.  For example.  I think that in the following situation we might close the client unintentionally.

```python
cluster = LocalCluster(processes=False)
client = Client(cluster)
cluster.scale(0)
assert client.status == "running"
```

We might want to only close the client if we created it (maybe add a new private variable in the `Worker._get_client` method).  

For testing we might create and close a worker and then verify that the attached client is closed.  This is not the best test, but maybe something like the following?

```python
async with Scheduler(...) as s:
    async with Worker(s.address) as w:
        await w.client
        assert w.client.status == "running"
    assert w.client.status == "closed"
```